### ISSUE TITLE
Possible problem in Sequencer backward on inputs of different lengths

### ISSUE THREAD

**Issue by boknilev** [Element-Research/rnn#76]
Hi,

I just updated the rnn package and am now having an error from the backward step in the Sequencer module. I have a network with some LSTMs in it, wrapped inside Sequencer modules. When I try to train  my network on inputs of variable length, I get this error:
Sequencer.lua:81: gradOutput should have as many elements as input

If I train my networks on inputs all the same length, I don't run into the error.

I wasn't able to replicate the problem in a simple contained network, so I cannot provide a minimal working example. However, this problem did not occur prior to update. 
I do notice that before updating, my network wasn't using Recursor modules (they did not exist in my version), and after updating it is using them. For example, if I add a non-recurrent module inside a Sequencer (e.g. Dropout), it gets printed with Recursor when I print the model (nn.Sequencer @ nn.Recurser @ nn.Dropout), whereas before updating it was not printed (nn.Sequencer @ nn.Dropout).

I know this is not a very detailed description, but do you have any idea what might be the source of this problem? 
I will keep debugging this.

**Comment by boknilev** [comment]
It appears that the problem was coming from a SelectTable module I had after the sequencer. I was trying to select the last element in the table, and had written a custom SelectTable to do that because at the time nn.SelectTable wasn't able to take negative indices. In my custom implementation I had forgotten to nullify self.gradInput for indices that are greater than the size of the input, which may have appeared from previous runs. After switching to nn.SplitTable(-1), I no longer have the error.

I'm still not sure why I wasn't getting this error before updating rnn though, but I guess it doesn't really matter so I'm closing this issue.
